modules/swanBidAdapter.js

```diff
@@ -0,0 +1,265 @@
+function getBaseRequest(bid, bidderRequest) {
+  let req = {
+    id: bidderRequest.auctionId,
+    cur: [config.getConfig('currency.adServerCurrency') || 'USD'],
+    at: 1,
+    tmax: config.getConfig('bidderTimeout'),
+    site: {
+      page: config.getConfig('pageUrl') || bidderRequest.refererInfo.referer
+    },
+    regs: {
+      coppa: config.getConfig('coppa') === true ? 1 : 0,
+    },
+    device: {
+      dnt: utils.getDNT() ? 1 : 0,
+      h: screen.height,
+      w: screen.width,
+      ua: window.navigator.userAgent,
+      language: window.navigator.language.split('-').shift()
+    }
+  };
+  if (bid.params.test) {
+    req.test = 1
+  }
+  if (bidderRequest.gdprConsent) {
+    if (bidderRequest.gdprConsent.gdprApplies !== undefined) {
+      utils.deepSetValue(req, 'regs.ext.gdpr', bidderRequest.gdprConsent.gdprApplies === true ? 1 : 0);
+    }
+    if (bidderRequest.gdprConsent.consentString !== undefined) {
+      utils.deepSetValue(req, 'user.ext.consent', bidderRequest.gdprConsent.consentString);
+    }
+    if (bidderRequest.gdprConsent.addtlConsent !== undefined) {
+      utils.deepSetValue(req, 'user.ext.ConsentedProvidersSettings.consented_providers', bidderRequest.gdprConsent.addtlConsent);
+    }
+  }
+  if (bidderRequest.uspConsent) {
+    utils.deepSetValue(req, 'regs.ext.us_privacy', bidderRequest.uspConsent);
+  }
+  if (bid.schain) {
+    utils.deepSetValue(req, 'source.ext.schain', bid.schain);
+  }
+  if (bid.userIdAsEids) {
+    utils.deepSetValue(req, 'user.ext.eids', bid.userIdAsEids);
+  }
+  return req;
+}
+
+function isVideoBid(bid) {
+  return utils.deepAccess(bid, 'mediaTypes.video');
+}
+
+function isBannerBid(bid) {
+  return utils.deepAccess(bid, 'mediaTypes.banner') || !isVideoBid(bid);
+}
+
+function getFloor(bid, mediaType) {
+  let floor = 0;
+
+  if (typeof bid.getFloor === 'function') {
+    const floorInfo = bid.getFloor({
+      currency: 'USD',
+      mediaType: mediaType,
+      size: '*'
+    });
+
+    if (typeof floorInfo === 'object' &&
+      floorInfo.currency === 'USD' &&
+      !isNaN(parseFloat(floorInfo.floor))) {
+      floor = Math.max(floor, parseFloat(floorInfo.floor));
+    }
+  }
+
+  return floor;
+}
+
+function interpretResponse(resp, req) {
+  const respBody = resp.body;
+  let imps = req.data.imp;
+
+  let bids = [];
+  respBody.seatbid.forEach(seatbid => {
+    bids = [...bids, ...seatbid.bid.map(bid => {
+      let response = {
+        requestId: bid.impid,
+        cpm: bid.price,
+        width: bid.w,
+        height: bid.h,
+        creativeId: bid.crid,
+        dealId: bid.dealid,
+        currency: respBody.cur || 'USD',
+        netRevenue: true,
+        ttl: 300,
+        mediaType: 'banner' in req.data.imp[0] ? BANNER : VIDEO,
+        meta: { advertiserDomains: bid.adomain }
+      };
+
+      let adm = bid.adm;
+      let imp = imps.find(i => i.id == bid.impid);
+      if (bid.ext && bid.ext.OWID) {
+        // if the ssp used swan data, add it to the tree
```

Comment:
if the ssp used the swan data they can respond with a tree which is appended to the swan offer stored globally